basestar-storage-elasticsearch/src/main/java/io/basestar/storage/elasticsearch/query/ESQueryStage.java

```diff
@@ -165,15 +165,23 @@ public Layout layout() {
                 last = mappings.fromSource(hit.getSourceAsMap());
                 results.add(last);
             }
-            final long total = hits.getTotalHits().value;
+            final long total = getTotal(token, hits);
             final Page.Token newPaging;
             if (total > results.size() && last != null) {
-                newPaging = KeysetPagingUtils.keysetPagingToken(schema, normalizedSort, last);
+                newPaging = KeysetPagingUtils.countPreservingKeysetPagingToken(schema, normalizedSort, last, total);
             } else {
                 newPaging = null;
             }
             return new Page<>(results, newPaging, Page.Stats.fromTotal(total));
         }
+
+        private long getTotal(Page.Token token, SearchHits hits) {
+            if (token != null) {
+                Map<String, List<Object>> objects = KeysetPagingUtils.countPreservingKeysetValues(schema, sort, token);
+                return (Long) objects.get("count").get(0);
```

Comment:
should we use `total` as a key (since `count` means how-many-to-take)?

Reply:
Can you define a helper class for the count preserving token info like:

```
@Data
public class CountPreservingTokenInfo {

   private final Long total;
   
   private final Page.Token token;
}
```

Reply:
added helper class but not sure if utils package is the best place for such class, @arcneon do you have a suggestion where to put such class?

Reply:
The `util` package is the right place I think